### PR TITLE
Np 45524 add facets to person query

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

**Comment by torarnet** [comment]
Will update Karate tests and OpenApi in later commit, everything else is ready for review.

File: cristin-commons/src/main/java/no/unit/nva/cristin/facet/FacetUriParamAppender.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        return CristinFacetKey.fromCristinFacet(cristinFacet);
+    }
+
+    private Map<String, List<String>> getQueryMap(String query) {
+        Map<String, List<String>> map = new HashMap<>();
+        Arrays.stream(query.split("&"))
+            .forEach(pair -> {
+                String[] keyValue = pair.split("=");
+                List<String> values = Arrays.asList(keyValue[1].split(","));
+                map.put(keyValue[0], new ArrayList<>(values));
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/facet/FacetUriParamAppender.java:79]: Name the '&', '=' and ','. The names should be something like QUERY_PARAMETER_DELIMITER, QUERY_PARAMETER_ASSIGNER and QUERY_VALUE_DELIMITER respectively.

File: cristin-commons/src/main/java/no/unit/nva/cristin/facet/FacetUriParamAppender.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            });
+        return map;
+    }
+
+    private Map<String, String> getCombinedMap(Map<String, List<String>> queryMap) {
+        var combinedMap = new TreeMap<String, String>();
+        for (Map.Entry<String, List<String>> entry : queryMap.entrySet()) {
+            var uniqueValues = new ArrayList<>(new HashSet<>(entry.getValue()));
+            Collections.sort(uniqueValues);
+            combinedMap.put(entry.getKey(), String.join(",", uniqueValues));
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/facet/FacetUriParamAppender.java:89]: I guess re-use the constant from above

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/query/CristinFacetKey.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public static Optional<CristinFacetKey> fromCristinFacet(CristinFacet cristinFacet) {
+        if (cristinFacet instanceof CristinSectorFacet) {
+            return Optional.of(CristinFacetKey.SECTOR);
+        } else if (cristinFacet instanceof CristinInstitutionFacet) {
+            return Optional.of(CristinFacetKey.INSTITUTION);
+        } else {
+            return Optional.empty();
+        }
```
Comment (brinxmat) [inline cristin-commons/src/main/java/no/unit/nva/cristin/model/query/CristinFacetKey.java:58]: Maybe have method in CristinFacet's extensions to return the value

File: cristin-commons/src/test/java/no/unit/nva/cristin/facet/FacetUriParamAppenderTest.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                         .getUriWithFacetKeys()
+                         .map(UriWrapper::getUri)
+                         .orElse(null);
+
+        assertEquals(idUriWithoutFacet, actual);
+    }
+
+    @Test
+    void shouldAppendSingleFacetValueToIdUri() {
+        CristinFacet cristinFacet = new CristinSectorFacet("UC", null);
```
Comment (brinxmat) [inline cristin-commons/src/test/java/no/unit/nva/cristin/facet/FacetUriParamAppenderTest.java:59]: `var` here and elsewhere

File: cristin-commons/src/test/java/no/unit/nva/cristin/facet/FacetUriParamAppenderTest.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import java.net.URI;
+import no.unit.nva.cristin.model.query.CristinFacet;
+import no.unit.nva.cristin.model.query.CristinInstitutionFacet;
+import no.unit.nva.cristin.model.query.CristinSectorFacet;
+import nva.commons.core.paths.UriWrapper;
+import org.junit.jupiter.api.Test;
+
+public class FacetUriParamAppenderTest {
```
Comment (brinxmat) [inline cristin-commons/src/test/java/no/unit/nva/cristin/facet/FacetUriParamAppenderTest.java:12]: package private

File: cristin-commons/src/test/java/no/unit/nva/cristin/facet/CristinFacetUriParamAppenderTest.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
+package no.unit.nva.cristin.facet;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CristinFacetUriParamAppenderTest {
```
Comment (brinxmat) [inline cristin-commons/src/test/java/no/unit/nva/cristin/facet/CristinFacetUriParamAppenderTest.java:10]: package private

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**